x-pack/legacy/plugins/maps/common/migrations/add_field_meta_options.test.js

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint max-len: 0 */
+
+import { addFieldMetaOptions } from './add_field_meta_options';
+import {  LAYER_TYPE, STYLE_TYPE } from '../constants';
+
+describe('addFieldMetaOptions', () => {
+
+  test('Should handle missing layerListJSON attribute', () => {
+    const attributes = {
+      title: 'my map',
+    };
+    expect(addFieldMetaOptions({ attributes })).toEqual({
+      title: 'my map',
+    });
+  });
+
+  test('Should ignore non-vector layers', () => {
+    const layerListJSON = JSON.stringify([
+      {
+        type: LAYER_TYPE.HEATMAP,
+        style: {
+          type: 'HEATMAP',
+          colorRampName: 'Greens'
+        }
+      }
+    ]);
+    const attributes = {
+      title: 'my map',
+      layerListJSON
+    };
+    expect(addFieldMetaOptions({ attributes })).toEqual({
+      title: 'my map',
+      layerListJSON
+    });
+  });
+
+  test('Should handle vector layer without style descriptor', () => {
+    const layerListJSON = JSON.stringify([
+      {
+        type: LAYER_TYPE.VECTOR,
+        style: null
+      }
+    ]);
+    const attributes = {
+      title: 'my map',
+      layerListJSON
+    };
+    expect(addFieldMetaOptions({ attributes })).toEqual({
+      title: 'my map',
+      layerListJSON
+    });
+  });
+
+  test('Should ignore static style properties', () => {
+    const layerListJSON = JSON.stringify([
+      {
+        type: LAYER_TYPE.VECTOR,
+        style: {
+          type: 'VECTOR',
+          properties: {
+            lineColor: {
+              type: STYLE_TYPE.STATIC,
+              options: {
+                color: '#FFFFFF'
+              }
+            }
+          }
+        }
+      }
+    ]);
+    const attributes = {
+      title: 'my map',
+      layerListJSON
+    };
+    expect(addFieldMetaOptions({ attributes })).toEqual({
+      title: 'my map',
+      layerListJSON
+    });
+  });
+
+  test('Should add field meta options to dynamic style properties', () => {
+    const layerListJSON = JSON.stringify([
+      {
+        type: LAYER_TYPE.VECTOR,
+        style: {
+          type: 'VECTOR',
+          properties: {
+            fillColor: {
+              type: STYLE_TYPE.DYNAMIC,
+              options: {
+                field: {
+                  name: 'my_field',
+                  origin: 'source'
+                },
+                color: 'Greys'
+              }
+            }
+          }
+        }
+      }
+    ]);
+    const attributes = {
+      title: 'my map',
+      layerListJSON
+    };
+    expect(addFieldMetaOptions({ attributes })).toEqual({
+      title: 'my map',
+      layerListJSON: '[{\"type\":\"VECTOR\",\"style\":{\"type\":\"VECTOR\",\"properties\":{\"fillColor\":{\"type\":\"DYNAMIC\",\"options\":{\"field\":{\"name\":\"my_field\",\"origin\":\"source\"},\"color\":\"Greys\",\"fieldMetaOptions\":{\"isEnabled\":false,\"sigma\":3}}}}}}]'
```

Comment:
this is difficult to read. could we format this as actual json, and use `JSON.stringify`?